### PR TITLE
feat: [CI-5283]: setting pod timeout for hosted/free users. 

### PR DESCRIPTION
**Pull request by HenAmar**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36463)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36463)

**Comment by HenAmar** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskParamsBuilder.java [hunk 1]
```diff
@@ -145,6 +152,26 @@ private CIK8InitializeTaskParams buildK8DirectTaskParams(InitializeStepInfo init
         .build();
   }
 
+  private Long getLimitTtl(NGAccess ngAccess) {
```
Comment (shubham149) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskParamsBuilder.java:155]: Move it to K8InitializeTaskUtils

**Review by shubham149 — COMMENTED** [review]
Have you tested any build that takes more than 1 hr with this change? What's the behaviour in terms of logs and build status?

**Comment by HenAmar** [comment]
> e? Wha

@shubham149 this is the output 
image

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskParamsBuilder.java [hunk 2]
```diff
@@ -92,7 +95,7 @@ public class K8InitializeTaskParamsBuilder {
   @Inject private HarnessImageUtils harnessImageUtils;
   @Inject private InternalContainerParamsProvider internalContainerParamsProvider;
   @Inject private SecretUtils secretUtils;
-
+  @Inject private NgLicenseHttpClient ngLicenseHttpClient;
```
Comment (shubham149) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskParamsBuilder.java:98]: Nit: remove it

**Review by shubham149 — APPROVED** [review]
Looks good to me

> Reviewed 2 of 6 files at r1, 5 of 6 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36463#-:-N9kQQc03dhrEQHmULjg:b-owaf1x)* status: :shipit: complete! ✓

**Review by shubham149 — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36463)* status: :shipit: complete! ✓

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
Nit: Is there a way to customize the error message to say evicted due to free tier usage limit and ask them to upgrade their license for longer executions?

> Reviewed 2 of 6 files at r1, 5 of 6 files at r2, 1 of 2 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36463#-:-N9mIWI961BZuO28czYH:b8r5kwd)* status: :shipit: complete! ✓

**Comment by HenAmar** [comment]
trigger codebasehash

**Pull request merged**